src/CTA.Rules.Models/Tokens/NodeToken.cs

```diff
@@ -51,5 +54,27 @@ public NodeToken()
 
         public NodeToken Clone() => (NodeToken)this.MemberwiseClone();
 
+        public List<GenericAction> AllActions
+        {
+            get
+            {
+                var allActions = new List<GenericAction>();
+                allActions.AddRange(AttributeActions);
+                allActions.AddRange(AttributeListActions);
+                allActions.AddRange(MethodDeclarationActions);
+                allActions.AddRange(ClassDeclarationActions);
+                allActions.AddRange(InterfaceDeclarationActions);
+                allActions.AddRange(ElementAccessActions);
+                allActions.AddRange(MemberAccessActions);
+                allActions.AddRange(IdentifierNameActions);
+                allActions.AddRange(InvocationExpressionActions);
+                allActions.AddRange(MemberAccessActions);
+                allActions.AddRange(UsingActions);
+                allActions.AddRange(ObjectCreationExpressionActions);
+                allActions.AddRange(NamespaceActions);
+                return allActions;
+            }
+        }
```

Comment:
I'm not sure I understand. How would this cause a memory leak?

Reply:
Memory leak might not be the best term here, but my thought is that it's possible to inadvertently consume more resources than expected. See the below example for what I was thinking, but if this is a nonconcern, we can forgo this modification.

Consider this trivial example:
``` csharp
var allActions1 = nodeToken.AllActions;
var allActions2 = nodeToken.AllActions;
```
With the current implementation, this would allocate 2x the memory than the suggested implementation:
```csharp
private List<GenericAction> _allActions;
public List<GenericAction> AllActions
{
    get
    {
        _allActions = new List<GenericAction>();
        _allActions.AddRange(AttributeActions);
        _allActions.AddRange(AttributeListActions);
        _allActions.AddRange(MethodDeclarationActions);
        _allActions.AddRange(ClassDeclarationActions);
        _allActions.AddRange(InterfaceDeclarationActions);
        _allActions.AddRange(ElementAccessActions);
        _allActions.AddRange(MemberAccessActions);
        _allActions.AddRange(IdentifierNameActions);
        _allActions.AddRange(InvocationExpressionActions);
        _allActions.AddRange(MemberAccessActions);
        _allActions.AddRange(UsingActions);
        _allActions.AddRange(ObjectCreationExpressionActions);
        _allActions.AddRange(NamespaceActions);
        return _allActions;
    }
}
```

Reply:
As discussed, this wouldn't allocate 2x memory because it's a local variable that gets disposed after the getter exits. We can look into this and the same behavior in fileactions to see if there's a better way to do it.